drud/ddev
# ddev and drush should use https URL when possible inside and outside container, fixes #1624, fixes #1612

## The Problem/Issue/Bug:
Drush commands should use the HTTPS hostname when possible, instead of HTTP.

## How this PR Solves The Problem:

* Stop using the $DDEV_URL environment variable
* Remove the /etc/drush/drushrc.php file from the container
* Use sites/default/drushrc.php (drush8) and sites/all/drush/drush.yml (drush9) to determine the uri for drush
* Abandon use of drush_ddev_settings.php and instead make the decisions it used to make inside settings.ddev.php
* Update tests
* Track the (believed) setup of mkcert during startup
* Use http or https URL based on mkcert status
* `ddev list`, `ddev start`, and `ddev describe` use the https URL when possible and show the http URL when mkcert is not usable.

## Manual Testing Instructions:

* Start ddev and `drush uli` both inside and outside container. Should give correct https URL
* `rm -rf $(mkcert -CAROOT)` to mess up mkcert. Start ddev. `drush uli` should work both inside and outside container, and give http url.
* Inspect `ddev start`, `ddev describe`, and `ddev list` for usage of appropriate URLs


## Automated Testing Overview:


## Related Issue Link(s):
#1624 OP
#1612: Request to only show appropriate http/https links in ddev start/ddev describe

## Release/Deployment notes:
